lint/rule_template_datasource.go

```diff
@@ -52,3 +100,33 @@ func getTemplateDatasource(d Dashboard) *Template {
 	}
 	return nil
 }
+
+func getTemplateDatasources(d Dashboard) []Template {
+
+	templateDatasources := []Template{}
+	for _, template := range d.Templating.List {
+		if template.Type != "datasource" {
+			continue
+		}
+		templateDatasources = append(templateDatasources, template)
+	}
+	return templateDatasources
+}
+
+// Returns datasource names that should be used instead of provided datasource
+func checkTemplatedDatasourceUsed(d Dashboard, datasource Datasource) []string {
+
+	datasource_count := len(getTemplateDatasources(d))
+	if datasource_count == 1 {
+		if datasource != "$datasource" && datasource != "${datasource}" {
+			return []string{"$datasource"}
+		}
+	} else if datasource_count == 2 {
```

Comment:
This could be `> 1` and I'd advise looping over the datasources, creating a slice containing `$<type>_datasource` and `${<type>_datasource}` then looping over them again to see if the *actual* name is contained in the slice.